File: src/main/java/org/eightsines/decompiled/A25.java
package org.eightsines.decompiled;

import kotlin.jvm.JvmField;

public final class A25 {
    private final double sq(double x) {
        return x * x;
    }

    private final boolean foo(double x, double y, double r, A25.Point p) {
        return sq(x - p.x) + sq(y - p.y) < sq(r);
    }

    private static final class Point {
        @JvmField
        public final double x;

        @JvmField
        public double y;

        public Point(double x, double y) {
            this.x = x;
            this.y = y;
        }
    }
}

File: src/main/kotlin/org/eightsines/kotlinperfopt/A25.kt
package org.eightsines.kotlinperfopt

class A25 {
    private class Point(@JvmField val x: Double, @JvmField var y: Double)

    private fun sq(x: Double) = x * x

    private fun foo(x: Double, y: Double, r: Double, p: Point) =
        sq(x - p.x) + sq(y - p.y) < sq(r)
}
